File: src/legacy-client/SkipAuthModal.js
var _this = this;
import { jsxDEV as _jsxDEV } from "react/jsx-dev-runtime";
import React from 'react';
var SkipAuthModal = function(param) {
    var onConfirm = param.onConfirm, onCancel = param.onCancel, onClose = param.onClose;
    return /*#__PURE__*/ _jsxDEV("div", {
        className: "modal-overlay",
        children: /*#__PURE__*/ _jsxDEV("div", {
            className: "modal-content",
            children: [
                /*#__PURE__*/ _jsxDEV("div", {
                    className: "modal-header",
                    children: [
                        /*#__PURE__*/ _jsxDEV("h2", {
                            className: "modal-title",
                            children: "Development Mode"
                        }, void 0, false, {
                            fileName: "SkipAuthModal.jsx",
                            lineNumber: 8,
                            columnNumber: 11
                        }, _this),
                        /*#__PURE__*/ _jsxDEV("button", {
                            onClick: onClose,
                            className: "modal-close-button",
                            children: "\xd7"
                        }, void 0, false, {
                            fileName: "SkipAuthModal.jsx",
                            lineNumber: 9,
                            columnNumber: 11
                        }, _this)
                    ]
                }, void 0, true, {
                    fileName: "SkipAuthModal.jsx",
                    lineNumber: 7,
                    columnNumber: 9
                }, _this),
                /*#__PURE__*/ _jsxDEV("div", {
                    className: "modal-body",
                    children: /*#__PURE__*/ _jsxDEV("p", {
                        children: "Use test data instead of logging in?"
                    }, void 0, false, {
                        fileName: "SkipAuthModal.jsx",
                        lineNumber: 12,
                        columnNumber: 11
                    }, _this)
                }, void 0, false, {
                    fileName: "SkipAuthModal.jsx",
                    lineNumber: 11,
                    columnNumber: 9
                }, _this),
                /*#__PURE__*/ _jsxDEV("div", {
                    className: "modal-actions modal-actions-center",
                    children: [
                        /*#__PURE__*/ _jsxDEV("button", {
                            onClick: onCancel,
                            className: "button-secondary",
                            children: "No, Log In"
                        }, void 0, false, {
                            fileName: "SkipAuthModal.jsx",
                            lineNumber: 15,
                            columnNumber: 11
                        }, _this),
                        /*#__PURE__*/ _jsxDEV("button", {
                            onClick: onConfirm,
                            className: "button-primary",
                            children: "Yes, Use Test Data"
                        }, void 0, false, {
                            fileName: "SkipAuthModal.jsx",
                            lineNumber: 18,
                            columnNumber: 11
                        }, _this)
                    ]
                }, void 0, true, {
                    fileName: "SkipAuthModal.jsx",
                    lineNumber: 14,
                    columnNumber: 9
                }, _this)
            ]
        }, void 0, true, {
            fileName: "SkipAuthModal.jsx",
            lineNumber: 6,
            columnNumber: 7
        }, _this)
    }, void 0, false, {
        fileName: "SkipAuthModal.jsx",
        lineNumber: 5,
        columnNumber: 5
    }, _this);
};
export default SkipAuthModal;

File: src/ios-client/views/SkipAuthModal.swift
import SwiftUI

struct SkipAuthModal: View {
    @Binding var isPresented: Bool
    var onConfirm: () -> Void
    var onCancel: () -> Void
    
    var body: some View {
        ZStack {
            Color.black.opacity(0.4)
                .ignoresSafeArea()
            VStack(spacing: 0) {
                // Header
                HStack {
                    Text("Development Mode")
                        .font(.headline)
                        .padding(.leading)
                    Spacer()
                    Button(action: { isPresented = false }) {
                        Image(systemName: "xmark")
                            .foregroundColor(.secondary)
                            .padding(8)
                    }
                    .accessibilityLabel("Close")
                }
                .padding(.vertical, 12)
                .background(Color(.systemBackground))
                Divider()
                // Body
                VStack(spacing: 24) {
                    Text("Use test data instead of logging in?")
                        .font(.body)
                        .multilineTextAlignment(.center)
                        .padding(.top, 24)
                    HStack(spacing: 16) {
                        Button(action: {
                            isPresented = false
                            onCancel()
                        }) {
                            Text("No, Log In")
                                .frame(maxWidth: .infinity)
                        }
                        .buttonStyle(.bordered)
                        Button(action: {
                            isPresented = false
                            onConfirm()
                        }) {
                            Text("Yes, Use Test Data")
                                .frame(maxWidth: .infinity)
                        }
                        .buttonStyle(.borderedProminent)
                    }
                }
                .padding([.horizontal, .bottom], 24)
                .background(Color(.systemBackground))
            }
            .frame(maxWidth: 340)
            .clipShape(RoundedRectangle(cornerRadius: 18, style: .continuous))
            .shadow(radius: 16)
        }
    }
}

#Preview {
    SkipAuthModal(isPresented: .constant(true), onConfirm: {}, onCancel: {})
} 